sk_sidekick/lib/src/commands/lock_dependencies_command.dart

```diff
@@ -0,0 +1,95 @@
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:yaml/yaml.dart';
+
+class LockDependenciesCommand extends Command {
+  @override
+  final String description = 'Locks all dependencies to their upper bounds';
+
+  @override
+  final String name = 'lock-dependencies';
+
+  @override
+  String get invocation => super.invocation.replaceFirst(
+        '[arguments]',
+        '[package-path]',
+      );
+
+  @override
+  Future<void> run() async {
+    final package = DartPackage.fromArgResults(argResults!);
+
+    final pubGet = systemDart(['pub', 'get'], workingDirectory: package.root);
```

Comment:
We should run this on CI with the latest dart version. Depending on the dart version we get different upper bounds. We can't guarantee that we use the latest version locally

Reply:
I'm checking that the Dart version used locally is the latest stable Dart version now, so we don't have to rely on CI for this: https://github.com/phntmxyz/sidekick/pull/160/files#diff-fb3ed57acb81967a5f3977b59c9842a7d8c82029a41ece4bb10d837c8b8ff227R38